# import Libraries

In [ ]:
import pandas as pd

In [ ]:
import numpy as np

# Load Dataset

In [ ]:
df = pd.read_csv("ESD.csv")
df

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
...,...,...,...,...,...,...,...,...,...,...,...,...,...
95,1096,2026-01-26,Aditya Verma,Kochi,South,Sports,Dumbbell Set,5,1999,5,9495.25,4.2,Credit Card
96,1097,2026-02-02,Arjun Nair,Mumbai,West,Clothing,T-Shirt,1,799,15,679.15,3.5,UPI
97,1098,2026-01-04,Yusuf Dar,Bengaluru,South,Home & Kitchen,Electric Kettle,4,1499,0,5996.00,4.2,Net Banking
98,1099,2026-03-06,Hiba Sheikh,Ahmedabad,West,Books,Machine Learning Basics,2,999,0,1998.00,3.8,UPI


# Data Selection

**Single Column Selection**






In [ ]:
df["Category"]

,Category
0,Electronics
1,Sports
2,Electronics
3,Electronics
4,Electronics
...,...
95,Sports
96,Clothing
97,Home & Kitchen
98,Books


**Select Multiple Columns**





In [ ]:
df[["Product","City","Total_Sales"]]

,Product,City,Total_Sales
0,Wireless Headphones,Hyderabad,4497.00
1,Running Shoes,Jammu,2799.00
2,Power Bank,Chennai,2038.30
3,Mechanical Keyboard,Lucknow,9496.20
4,Smart Watch,Jammu,3134.05
...,...,...,...
95,Dumbbell Set,Kochi,9495.25
96,T-Shirt,Mumbai,679.15
97,Electric Kettle,Bengaluru,5996.00
98,Machine Learning Basics,Ahmedabad,1998.00


**Selecting Rows Using** **iloc**

In [ ]:
df.iloc[1:5]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI


In [ ]:
df.iloc[0:10,1:6] # Specific rows and columns

,Order_Date,Customer,City,Region,Category
0,2026-02-27,Kabir Ali,Hyderabad,South,Electronics
1,2026-01-24,Aditya Verma,Jammu,North,Sports
2,2026-04-18,Sara Ahmed,Chennai,South,Electronics
3,2026-03-13,Kabir Ali,Lucknow,North,Electronics
4,2026-03-30,Priya Menon,Jammu,North,Electronics
5,2026-01-21,Manya Rao,Kochi,South,Books
6,2026-01-12,Alina Shah,Jaipur,North,Home & Kitchen
7,2026-04-27,Aman Kumar,Jammu,North,Clothing
8,2026-06-05,Aman Kumar,Mumbai,West,Home & Kitchen
9,2026-03-11,Aman Kumar,Kochi,South,Clothing


**Using Loc**

In [ ]:
df.loc[0:6,["Customer","City","Product"]]

,Customer,City,Product
0,Kabir Ali,Hyderabad,Wireless Headphones
1,Aditya Verma,Jammu,Running Shoes
2,Sara Ahmed,Chennai,Power Bank
3,Kabir Ali,Lucknow,Mechanical Keyboard
4,Priya Menon,Jammu,Smart Watch
5,Manya Rao,Kochi,The Alchemist
6,Alina Shah,Jaipur,Water Bottle


# **Filtering Operations**

In [ ]:
#Orders With Sales Greater Than 5000
high_sales = df[df["Total_Sales"] > 5000]
high_sales

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
7,1008,2026-04-27,Aman Kumar,Jammu,North,Clothing,Hoodie,4,1599,5,6076.20,4.3,Credit Card
9,1010,2026-03-11,Aman Kumar,Kochi,South,Clothing,Jacket,4,2499,10,8996.40,4.7,Credit Card
10,1011,2026-03-22,Rahul Das,Hyderabad,South,Home & Kitchen,Coffee Maker,2,3499,0,6998.00,4.2,UPI
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
13,1014,2026-02-05,Dev Patel,Jaipur,North,Sports,Running Shoes,3,2799,0,8397.00,4.6,UPI
16,1017,2026-03-17,Sana Malik,Hyderabad,South,Sports,Cricket Bat,3,2499,0,7497.00,4.0,Net Banking
18,1019,2026-05-19,Reyansh Jain,Chennai,South,Clothing,Jacket,3,2499,0,7497.00,4.7,UPI
27,1028,2026-04-19,Sana Malik,Chandigarh,North,Electronics,Wireless Headphones,4,1499,5,5696.20,4.6,UPI
28,1029,2026-01-28,Sara Ahmed,Hyderabad,South,Electronics,Wireless Headphones,4,1499,5,5696.20,4.1,Credit Card


In [ ]:
#Orders With Quantity Greater Than and equals to 4
df[df["Quantity"] >= 4]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
7,1008,2026-04-27,Aman Kumar,Jammu,North,Clothing,Hoodie,4,1599,5,6076.20,4.3,Credit Card
9,1010,2026-03-11,Aman Kumar,Kochi,South,Clothing,Jacket,4,2499,10,8996.40,4.7,Credit Card
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
15,1016,2026-03-06,Manya Rao,Pune,West,Books,The Alchemist,4,499,15,1696.60,3.5,UPI
24,1025,2026-01-02,Rohan Mehta,Chennai,South,Sports,Football,5,799,0,3995.00,5.0,UPI
26,1027,2026-05-02,Sara Ahmed,Chandigarh,North,Sports,Football,5,799,15,3395.75,4.6,Net Banking
27,1028,2026-04-19,Sana Malik,Chandigarh,North,Electronics,Wireless Headphones,4,1499,5,5696.20,4.6,UPI
28,1029,2026-01-28,Sara Ahmed,Hyderabad,South,Electronics,Wireless Headphones,4,1499,5,5696.20,4.1,Credit Card
29,1030,2026-03-13,Karan Joshi,Bengaluru,South,Books,Python Programming,4,749,0,2996.00,4.1,UPI


In [ ]:
# Sports Orders
df[df["Category"] == "Sports"]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
11,1012,2026-04-12,Aman Kumar,Chandigarh,North,Sports,Cricket Bat,2,2499,10,4498.20,4.6,Credit Card
13,1014,2026-02-05,Dev Patel,Jaipur,North,Sports,Running Shoes,3,2799,0,8397.00,4.6,UPI
16,1017,2026-03-17,Sana Malik,Hyderabad,South,Sports,Cricket Bat,3,2499,0,7497.00,4.0,Net Banking
20,1021,2026-01-18,Reyansh Jain,Chandigarh,North,Sports,Yoga Mat,1,899,10,809.10,4.7,Credit Card
22,1023,2026-06-21,Aman Kumar,Jaipur,North,Sports,Football,3,799,10,2157.30,4.3,Net Banking
23,1024,2026-02-27,Rohan Mehta,Jaipur,North,Sports,Running Shoes,1,2799,0,2799.00,4.3,UPI
24,1025,2026-01-02,Rohan Mehta,Chennai,South,Sports,Football,5,799,0,3995.00,5.0,UPI
26,1027,2026-05-02,Sara Ahmed,Chandigarh,North,Sports,Football,5,799,15,3395.75,4.6,Net Banking
30,1031,2026-01-04,Rohan Mehta,Ahmedabad,West,Sports,Yoga Mat,1,899,15,764.15,4.1,Credit Card


In [ ]:
#Orders Paid Using Net Banking
df[df["Payment_Method"] == "Net Banking"]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
16,1017,2026-03-17,Sana Malik,Hyderabad,South,Sports,Cricket Bat,3,2499,0,7497.00,4.0,Net Banking
22,1023,2026-06-21,Aman Kumar,Jaipur,North,Sports,Football,3,799,10,2157.30,4.3,Net Banking
25,1026,2026-05-12,Sara Ahmed,Chandigarh,North,Electronics,Wireless Headphones,3,1499,0,4497.00,4.2,Net Banking
26,1027,2026-05-02,Sara Ahmed,Chandigarh,North,Sports,Football,5,799,15,3395.75,4.6,Net Banking
32,1033,2026-06-28,Maryam Khan,Pune,West,Home & Kitchen,Mixer Grinder,3,2999,10,8097.30,4.7,Net Banking
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.00,4.0,Net Banking
49,1050,2026-03-10,Ishita Gupta,Srinagar,North,Clothing,T-Shirt,4,799,0,3196.00,3.8,Net Banking
53,1054,2026-04-22,Aisha Mir,Chennai,South,Home & Kitchen,Coffee Maker,1,3499,15,2974.15,4.3,Net Banking
54,1055,2026-02-18,Arjun Nair,Chennai,South,Sports,Yoga Mat,4,899,15,3056.60,3.5,Net Banking


In [ ]:
# Multiple Conditions
df[(df["Quantity"] == 5) & (df["Category"] == "Electronics")]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
46,1047,2026-02-07,Sana Malik,Bengaluru,South,Electronics,Wireless Mouse,5,899,15,3820.75,4.0,UPI
83,1084,2026-06-19,Aisha Mir,Kolkata,East,Electronics,Wireless Mouse,5,899,10,4045.50,3.8,Cash on Delivery


In [ ]:
# Orders With Low Quantity and low Sales
df[(df["Quantity"] < 2) & (df["Total_Sales"]< 500)]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
5,1006,2026-01-21,Manya Rao,Kochi,South,Books,The Alchemist,1,499,10,449.1,4.2,Cash on Delivery
62,1063,2026-04-17,Aman Kumar,Chandigarh,North,Books,The Alchemist,1,499,10,449.1,4.8,Credit Card


# **Sorting Operations**

In [ ]:
# Sort Orders by Category
df.sort_values(by="Category",ascending = True)

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
99,1100,2026-02-08,Kabir Ali,Pune,West,Books,Data Science Handbook,5,899,0,4495.0,4.8,Debit Card
44,1045,2026-03-27,Reyansh Jain,Jammu,North,Books,Machine Learning Basics,1,999,0,999.0,4.0,Debit Card
45,1046,2026-01-03,Ananya Singh,Delhi,North,Books,The Alchemist,4,499,15,1696.6,3.8,Credit Card
98,1099,2026-03-06,Hiba Sheikh,Ahmedabad,West,Books,Machine Learning Basics,2,999,0,1998.0,3.8,UPI
21,1022,2026-05-16,Priya Menon,Delhi,North,Books,The Alchemist,2,499,5,948.1,4.5,Credit Card
...,...,...,...,...,...,...,...,...,...,...,...,...,...
71,1072,2026-05-03,Aarav Sharma,Jaipur,North,Sports,Yoga Mat,5,899,10,4045.5,4.3,Debit Card
13,1014,2026-02-05,Dev Patel,Jaipur,North,Sports,Running Shoes,3,2799,0,8397.0,4.6,UPI
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.0,4.0,Net Banking
24,1025,2026-01-02,Rohan Mehta,Chennai,South,Sports,Football,5,799,0,3995.0,5.0,UPI


In [ ]:
# Top 5 Highest-Selling Orders
df.sort_values(by="Total_Sales",ascending = False).head(5)

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.00,4.0,Net Banking
52,1053,2026-01-08,Ananya Singh,Hyderabad,South,Sports,Running Shoes,5,2799,5,13295.25,4.2,Credit Card
84,1085,2026-01-28,Sana Malik,Hyderabad,South,Home & Kitchen,Coffee Maker,4,3499,10,12596.40,4.3,Net Banking
89,1090,2026-05-13,Aditya Verma,Lucknow,North,Sports,Cricket Bat,5,2499,0,12495.00,3.8,Credit Card
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking


In [ ]:
#Sort Product by rating
df.sort_values(by="Rating", ascending = False).head(3)


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
24,1025,2026-01-02,Rohan Mehta,Chennai,South,Sports,Football,5,799,0,3995.0,5.0,UPI
35,1036,2026-06-02,Rohan Mehta,Srinagar,North,Sports,Yoga Mat,2,899,0,1798.0,5.0,Credit Card
47,1048,2026-06-24,Sara Ahmed,Lucknow,North,Home & Kitchen,Coffee Maker,3,3499,0,10497.0,5.0,UPI


In [ ]:
# Sort by payment method
df.sort_values(by="Payment_Method",ascending = True).head(10)

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
85,1086,2026-06-16,Arjun Nair,Kochi,South,Clothing,Jeans,2,1899,15,3228.30,4.8,Cash on Delivery
65,1066,2026-06-20,Dev Patel,Jaipur,North,Electronics,Smart Watch,2,3299,10,5938.20,4.7,Cash on Delivery
82,1083,2026-06-25,Priya Menon,Jammu,North,Sports,Football,1,799,5,759.05,3.8,Cash on Delivery
78,1079,2026-03-05,Aditya Verma,Mumbai,West,Clothing,Hoodie,1,1599,0,1599.00,3.5,Cash on Delivery
60,1061,2026-05-12,Alina Shah,Bengaluru,South,Books,Python Programming,1,749,5,711.55,5.0,Cash on Delivery
40,1041,2026-03-14,Ishita Gupta,Delhi,North,Home & Kitchen,Coffee Maker,2,3499,5,6648.10,4.6,Cash on Delivery
92,1093,2026-05-16,Aisha Mir,Hyderabad,South,Sports,Cricket Bat,4,2499,15,8496.60,4.5,Cash on Delivery
17,1018,2026-01-28,Aman Kumar,Srinagar,North,Home & Kitchen,Water Bottle,2,699,15,1188.30,4.2,Cash on Delivery
83,1084,2026-06-19,Aisha Mir,Kolkata,East,Electronics,Wireless Mouse,5,899,10,4045.50,3.8,Cash on Delivery


# **Basic Sales** **Statistics**

In [ ]:
#Total Sales
total_sales = df["Total_Sales"].sum()
print("Total Sales:",total_sales)

Total Sales: 447262.95000000007


In [ ]:
# Average Sales
avg_sales = df["Total_Sales"].mean()
print("Average Sales:",avg_sales)

Average Sales: 4472.629500000001


In [ ]:
#Highest Sales
highest_sales = df["Total_Sales"].max()
print("highest sales:",highest_sales)

highest sales: 13995.0


In [ ]:
# Lowest Sales
lowest_sales = df["Total_Sales"].min()
print("Lowest Sales:",lowest_sales)

Lowest Sales: 449.1


In [ ]:
# Total Quantity Sold
total_quantity = df["Quantity"].sum()
print("Total Quantity Sold",total_quantity)

Total Quantity Sold 294


In [ ]:
# Numbers of Orders
numbers_of_orders = df["Order_ID"].nunique()
print("Numbers of Orders:",numbers_of_orders)

Numbers of Orders: 100


# **GroupBy Analysis**

In [ ]:
# Sales by Category
category_sales = df.groupby("Category")["Total_Sales"].sum()
category_sales

,Total_Sales
Category,
Books,36792.40
Clothing,69028.20
Electronics,92086.00
Home & Kitchen,113329.90
Sports,136026.45


In [ ]:
# Average Sales by Category
df.groupby("Category")["Total_Sales"].mean().sort_values(ascending=False)

,Total_Sales
Category,
Sports,5441.058000
Home & Kitchen,5396.661905
Clothing,4930.585714
Electronics,4846.631579
Books,1752.019048


In [ ]:
# Quantity Sold by Category
df.groupby("Category")["Order_ID"].count().sort_values(ascending=False)

,Order_ID
Category,
Sports,25
Books,21
Home & Kitchen,21
Electronics,19
Clothing,14


In [ ]:
# Number of Orders by Category
df.groupby("Category")["Order_ID"].count().sort_values(ascending = False)

,Order_ID
Category,
Sports,25
Books,21
Home & Kitchen,21
Electronics,19
Clothing,14


# **Aggregate Functions**

In [ ]:
category_analysis = df.groupby("Category").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Highest_Sales=("Total_Sales", "max"),
    Lowest_Sales=("Total_Sales", "min"),
    Total_Quantity=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count"))
category_analysis.sort_values("Total_Sales", ascending=False)

,Total_Sales,Average_Sales,Highest_Sales,Lowest_Sales,Total_Quantity,Number_of_Orders
Category,,,,,,
Sports,136026.45,5441.058000,13995.00,759.05,79,25
Home & Kitchen,113329.90,5396.661905,12596.40,1188.30,59,21
Electronics,92086.00,4846.631579,9496.20,1708.10,57,19
Clothing,69028.20,4930.585714,10620.75,679.15,39,14
Books,36792.40,1752.019048,4495.00,449.10,60,21


# **City Analysis**

In [ ]:
# Total Sales by City
city_sales = df.groupby("City")["Total_Sales"].sum().sort_values(ascending=False)
print("City Sales:",city_sales)

City Sales: City
Hyderabad      59076.45
Chennai        47493.20
Lucknow        46194.65
Pune           41908.55
Mumbai         41457.80
Kochi          37257.50
Chandigarh     34539.65
Delhi          30882.80
Jaipur         25534.00
Kolkata        20169.00
Jammu          16059.75
Ahmedabad      14788.70
Bengaluru      13524.30
Srinagar       11778.60
Bhubaneswar     6598.00
Name: Total_Sales, dtype: float64


In [ ]:
# Average Sales by City
Average_sales = df.groupby("City")["Total_Sales"].mean().sort_values(ascending=False)
print("Average Sales by City:",Average_sales)

Average Sales by City: City
Hyderabad      8439.492857
Lucknow        7699.108333
Bhubaneswar    6598.000000
Mumbai         5182.225000
Delhi          5147.133333
Chennai        4749.320000
Chandigarh     4317.456250
Pune           4190.855000
Kochi          4139.722222
Jaipur         3647.714286
Bengaluru      3381.075000
Kolkata        2881.285714
Jammu          2676.625000
Ahmedabad      2464.783333
Srinagar       2355.720000
Name: Total_Sales, dtype: float64


In [ ]:
# Quantity Sold by City
df.groupby("City")["Quantity"].sum().sort_values(ascending=False)

,Quantity
City,
Pune,34
Kochi,29
Chennai,29
Hyderabad,25
Kolkata,24
Mumbai,23
Chandigarh,22
Lucknow,22
Delhi,18


# **Product Analysis**

In [ ]:
# Total Sales by Product
product_sales = df.groupby("Product")["Total_Sales"].sum().sort_values(ascending=False)
product_sales

,Total_Sales
Product,
Coffee Maker,51610.25
Jacket,47106.15
Cricket Bat,41483.40
Running Shoes,41285.25
Wireless Headphones,33877.40
Dumbbell Set,30184.90
Mechanical Keyboard,26489.40
Electric Kettle,26382.40
Mixer Grinder,25341.55


In [ ]:
# Top 10 Products by Sales
top_products = (
    df.groupby("Product")["Total_Sales"].sum().sort_values(ascending=False).head(10))
top_products

,Total_Sales
Product,
Coffee Maker,51610.25
Jacket,47106.15
Cricket Bat,41483.40
Running Shoes,41285.25
Wireless Headphones,33877.40
Dumbbell Set,30184.90
Mechanical Keyboard,26489.40
Electric Kettle,26382.40
Mixer Grinder,25341.55


In [ ]:
# Products With Highest Quantity Sold
df.groupby("Product")["Quantity"].sum().sort_values(ascending=False).head(10)

,Quantity
Product,
Wireless Headphones,23
Atomic Habits,21
Jacket,20
Electric Kettle,19
Cricket Bat,18
The Alchemist,17
Coffee Maker,16
Dumbbell Set,16
Yoga Mat,16


# **Payment Method Analysis**

In [ ]:
# Number of Orders by Payment Method
df["Payment_Method"].value_counts()

,count
Payment_Method,
UPI,26
Credit Card,22
Debit Card,20
Net Banking,17
Cash on Delivery,15


In [ ]:
# Total Sales by Payment Method
payment_sales = df.groupby("Payment_Method")["Total_Sales"].sum()
payment_sales.sort_values(ascending=False)

,Total_Sales
Payment_Method,
Credit Card,106540.25
Debit Card,98064.90
Net Banking,96371.25
UPI,96163.25
Cash on Delivery,50123.30


In [ ]:
# Average Sales by Payment Method
df.groupby("Payment_Method")["Total_Sales"].mean().sort_values(ascending=False)

,Total_Sales
Payment_Method,
Net Banking,5668.897059
Debit Card,4903.245000
Credit Card,4842.738636
UPI,3698.586538
Cash on Delivery,3341.553333


# **Finding Specific Orders**

In [ ]:
# Highest-Selling Order
highest_order = df.loc[df["Total_Sales"].idxmax()]
highest_order

,48
Order_ID,1049
Order_Date,2026-03-02
Customer,Ishita Gupta
City,Mumbai
Region,West
Category,Sports
Product,Running Shoes
Quantity,5
Unit_Price,2799
Discount_Percent,0


In [ ]:
# Lowest-Selling Order
lowest_order = df.loc[df["Total_Sales"].idxmin()]
lowest_order

,5
Order_ID,1006
Order_Date,2026-01-21
Customer,Manya Rao
City,Kochi
Region,South
Category,Books
Product,The Alchemist
Quantity,1
Unit_Price,499
Discount_Percent,10


In [ ]:
# Top 5 Orders
df.nlargest(5, "Total_Sales")

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.00,4.0,Net Banking
52,1053,2026-01-08,Ananya Singh,Hyderabad,South,Sports,Running Shoes,5,2799,5,13295.25,4.2,Credit Card
84,1085,2026-01-28,Sana Malik,Hyderabad,South,Home & Kitchen,Coffee Maker,4,3499,10,12596.40,4.3,Net Banking
89,1090,2026-05-13,Aditya Verma,Lucknow,North,Sports,Cricket Bat,5,2499,0,12495.00,3.8,Credit Card
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking


In [ ]:
# Orders With Discount Greater Than 10%
df[df["Discount_Percent"] > 13]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
15,1016,2026-03-06,Manya Rao,Pune,West,Books,The Alchemist,4,499,15,1696.60,3.5,UPI
17,1018,2026-01-28,Aman Kumar,Srinagar,North,Home & Kitchen,Water Bottle,2,699,15,1188.30,4.2,Cash on Delivery
26,1027,2026-05-02,Sara Ahmed,Chandigarh,North,Sports,Football,5,799,15,3395.75,4.6,Net Banking
30,1031,2026-01-04,Rohan Mehta,Ahmedabad,West,Sports,Yoga Mat,1,899,15,764.15,4.1,Credit Card
34,1035,2026-05-03,Dev Patel,Kochi,South,Home & Kitchen,Coffee Maker,1,3499,15,2974.15,4.7,Credit Card
36,1037,2026-06-02,Zoya Khan,Pune,West,Electronics,Power Bank,2,1199,15,2038.30,4.8,UPI
45,1046,2026-01-03,Ananya Singh,Delhi,North,Books,The Alchemist,4,499,15,1696.60,3.8,Credit Card
46,1047,2026-02-07,Sana Malik,Bengaluru,South,Electronics,Wireless Mouse,5,899,15,3820.75,4.0,UPI


In [ ]:
# Orders With Rating less than 2.5
df[df["Rating"] < 4]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
15,1016,2026-03-06,Manya Rao,Pune,West,Books,The Alchemist,4,499,15,1696.60,3.5,UPI
38,1039,2026-04-28,Vivaan Kapoor,Chandigarh,North,Home & Kitchen,Mixer Grinder,2,2999,5,5698.10,3.8,UPI
45,1046,2026-01-03,Ananya Singh,Delhi,North,Books,The Alchemist,4,499,15,1696.60,3.8,Credit Card
49,1050,2026-03-10,Ishita Gupta,Srinagar,North,Clothing,T-Shirt,4,799,0,3196.00,3.8,Net Banking
54,1055,2026-02-18,Arjun Nair,Chennai,South,Sports,Yoga Mat,4,899,15,3056.60,3.5,Net Banking
61,1062,2026-02-20,Kabir Ali,Mumbai,West,Electronics,Mechanical Keyboard,3,2499,0,7497.00,3.5,UPI
63,1064,2026-02-07,Aman Kumar,Kochi,South,Books,Atomic Habits,4,599,0,2396.00,3.5,UPI
69,1070,2026-06-30,Aditya Verma,Delhi,North,Clothing,Jacket,4,2499,0,9996.00,3.8,Net Banking
75,1076,2026-04-29,Ishita Gupta,Chennai,South,Electronics,Wireless Headphones,4,1499,0,5996.00,3.5,Debit Card


In [ ]:
# Orders Paid Through UPI With Sales below 2500
df[(df["Payment_Method"] == "UPI") & (df["Total_Sales"] <2500)]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
15,1016,2026-03-06,Manya Rao,Pune,West,Books,The Alchemist,4,499,15,1696.60,3.5,UPI
31,1032,2026-01-16,Ishita Gupta,Pune,West,Books,Atomic Habits,2,599,10,1078.20,4.5,UPI
33,1034,2026-05-29,Maryam Khan,Delhi,North,Clothing,T-Shirt,2,799,0,1598.00,4.7,UPI
36,1037,2026-06-02,Zoya Khan,Pune,West,Electronics,Power Bank,2,1199,15,2038.30,4.8,UPI
43,1044,2026-05-10,Hiba Sheikh,Chennai,South,Clothing,T-Shirt,2,799,5,1518.10,4.2,UPI
63,1064,2026-02-07,Aman Kumar,Kochi,South,Books,Atomic Habits,4,599,0,2396.00,3.5,UPI
68,1069,2026-03-27,Vivaan Kapoor,Ahmedabad,West,Electronics,Wireless Mouse,2,899,5,1708.10,4.7,UPI
77,1078,2026-01-05,Maryam Khan,Kolkata,East,Books,Machine Learning Basics,1,999,10,899.10,4.7,UPI
86,1087,2026-03-10,Vivaan Kapoor,Kolkata,East,Books,Atomic Habits,4,599,15,2036.60,4.1,UPI
96,1097,2026-02-02,Arjun Nair,Mumbai,West,Clothing,T-Shirt,1,799,15,679.15,3.5,UPI


In [ ]:
# Sales by Category and Payment Method
category_payment = df.groupby(
    ["Category", "Payment_Method"])["Total_Sales"].sum()
category_payment

Category        Payment_Method  
Books           Cash on Delivery     1160.65
                Credit Card          6912.85
                Debit Card           9838.80
                Net Banking          5779.60
                UPI                 13100.50
Clothing        Cash on Delivery     8055.60
                Credit Card         15072.60
                Debit Card          10795.00
                Net Banking         23812.75
                UPI                 11292.25
Electronics     Cash on Delivery    16519.00
                Credit Card         12294.20
                Debit Card          31883.40
                Net Banking          4497.00
                UPI                 26892.40
Home & Kitchen  Cash on Delivery    12333.40
                Credit Card         12919.30
                Debit Card          26209.85
                Net Banking         32180.25
                UPI                 29687.10
Sports          Cash on Delivery    12054.65
                Credit Card         59341.30
                Debit Card          19337.85
                Net Banking         30101.65
                UPI                 15191.00
Name: Total_Sales, dtype: float64

In [ ]:
# Sales by City and Category
city_category = df.groupby(
    ["City", "Category"])["Total_Sales"].sum()
city_category

City         Category      
Ahmedabad    Books              1998.00
             Electronics        6205.10
             Home & Kitchen     3822.45
             Sports             2763.15
Bengaluru    Books              3707.55
             Electronics        3820.75
             Home & Kitchen     5996.00
Bhubaneswar  Electronics        6598.00
Chandigarh   Books               449.10
             Electronics       19689.40
             Home & Kitchen     5698.10
             Sports             8703.05
Chennai      Clothing           9015.10
             Electronics        8034.30
             Home & Kitchen    14895.60
             Sports            15548.20
Delhi        Books              2644.70
             Clothing          21590.00
             Home & Kitchen     6648.10
Hyderabad    Electronics       10193.20
             Home & Kitchen    19594.40
             Sports            29288.85
Jaipur       Clothing            799.00
             Electronics        5938.20
             Home & Kitchen     1398.00
             Sports            17398.80
Jammu        Books               999.00
             Clothing           6076.20
             Electronics        3134.05
             Sports             5850.50
Kochi        Books              5840.10
             Clothing          15453.00
             Home & Kitchen     6469.15
             Sports             9495.25
Kolkata      Books              6754.75
             Electronics        7043.50
             Home & Kitchen     6370.75
Lucknow      Books              3085.70
             Clothing          10620.75
             Electronics        9496.20
             Home & Kitchen    10497.00
             Sports            12495.00
Mumbai       Books              2245.50
             Clothing           2278.15
             Electronics        9895.00
             Home & Kitchen    13044.15
             Sports            13995.00
Pune         Books              7269.80
             Electronics        2038.30
             Home & Kitchen    17707.90
             Sports            14892.55
Srinagar     Books              1798.20
             Clothing           3196.00
             Home & Kitchen     1188.30
             Sports             5596.10
Name: Total_Sales, dtype: float64

## Meaningful Observations

1. The dataset contains **100 orders** with a total sales value of **₹447,262.95**.

2. The average sales value per order is approximately **₹4,472.63**, while the highest individual order generated **₹13,995** in sales.

3. **Sports** is the highest-performing category with total sales of approximately **₹136,026.45**, followed by **Home & Kitchen** with approximately **₹113,329.90**.

4. **Hyderabad** generated the highest total sales among the cities, with approximately **₹59,076.45**.

5. **Coffee Maker** is the top-performing product by total sales, generating approximately **₹51,610.25**.

6. **UPI** is the most frequently used payment method, accounting for **26 orders**, while Cash on Delivery has the fewest orders.

7. The dataset records a total quantity of **294 products sold** across the 100 orders.

8. The analysis shows that grouping and aggregation by category, city, product, and payment method helps identify the strongest-performing areas of the e-commerce business.
